demonstrations/tutorial_qubit_tapering.py

```diff
@@ -0,0 +1,273 @@
+# applying the Hamiltonians to the Hartree-Fock state. For the tapered Hamiltonian, this requires
+# transforming the Hartree-Fock state with the same symmetries obtained for the original
+# Hamiltonian. This reduces the number of qubits in the Hartree-Fock state to match that of the
+# tapered Hamiltonian. It can be done with the :func:`~.pennylane.qchem.taper_hf`.
+
+state_tapered = qml.qchem.taper_hf(
+                generators, paulixops, paulix_sector, n_electrons, len(H.wires))
+print(state_tapered)
+
+##############################################################################
+# Recall that the original Hartree-Fock state for the :math:`\textrm{HeH}^+` cation is
+# :math:`[1 1 0 0]`. We can now generate the qubit representation of these states and compute the
+# Hartree-Fock energies for each Hamiltonian
+
+dev = qml.device('default.qubit', wires=H.wires)
+@qml.qnode(dev)
+def circuit():
+    qml.BasisState(np.array([1, 1, 0, 0]), wires=H.wires)
+    return qml.state()
+qubit_state = circuit()
+HF_energy = qubit_state.T @ qml.utils.sparse_hamiltonian(H).toarray() @ qubit_state
+print(f'HF energy: {np.real(HF_energy):.8f} Ha')
+
+dev = qml.device('default.qubit', wires=H_tapered.wires)
+@qml.qnode(dev)
+def circuit():
+    qml.BasisState(np.array([1, 1]), wires=H_tapered.wires)
+    return qml.state()
+qubit_state = circuit()
+HF_energy = qubit_state.T @ qml.utils.sparse_hamiltonian(H_tapered).toarray() @ qubit_state
+print(f'HF energy (tapered): {np.real(HF_energy):.8f} Ha')
+
+##############################################################################
+# These values are identical to the reference Hartree-Fock energy :math:`-2.8543686493` Ha.
+#
+# VQE simulation
+# --------------
+# Finally, we can use the tapered Hamiltonian and the tapered references state to perform a VQE
+# simulation and compute the ground state energy of the :math:`\textrm{HeH}^+` cation. We use the
+# tapered Hartree-Fock state to build a circuit that prepares an entangled state by applying Pauli
+# rotation gates [#ryabinkin2018]_ since we cannot use the typical particle-conserving gates
+# with the tapered state
+
+dev = qml.device('default.qubit', wires=H_tapered.wires)
+@qml.qnode(dev)
+def circuit(params):
+    qml.BasisState(state_tapered, wires=H_tapered.wires)
+    qml.PauliRot(params[2], 'Y',  wires=[0])
+    qml.PauliRot(params[1], 'Y',  wires=[1])
+    qml.PauliRot(params[0], 'YX', wires=[0, 1])
+    return qml.expval(H_tapered)
+
+##############################################################################
+# We define an optimizer and the initial values of the circuit parameters and optimize the circuit
+# parameters with respect to the ground state energy
+
+optimizer = qml.GradientDescentOptimizer(stepsize=0.5)
+params = np.zeros(3)
+
+for n in range(1, 21):
+    params, energy = optimizer.step_and_cost(circuit, params)
+    print(f'n: {n}, E: {energy:.8f} Ha, Params: {params}')

```

Comment:
```suggestion
    if n%2:
        print(f'n: {n}, E: {energy:.8f} Ha, Params: {params}')

```
Will reduce the block length.